### PR TITLE
test: move common logic into TestBase

### PR DESCRIPTION
**Pull request by vbarua**

### CODE REVIEW

**Comment by vbarua** [comment]
Reducing test code duplication by moving more common logic into the TestBase abstract class.

File: core/src/test/java/io/substrait/type/proto/AggregateRoundtripTest.java [hunk 1]
```diff
@@ -21,8 +21,6 @@
 import org.junit.jupiter.api.Test;
 
 public class AggregateRoundtripTest extends TestBase {
-  static final org.slf4j.Logger logger =
-      org.slf4j.LoggerFactory.getLogger(AggregateRoundtripTest.class);
```
Comment (vbarua) [inline core/src/test/java/io/substrait/type/proto/AggregateRoundtripTest.java:25]: These loggers aren't used or needed so removing them to reduce noise.

File: core/src/test/java/io/substrait/type/proto/LocalFilesRoundtripTest.java [hunk 2]
```diff
@@ -25,15 +25,7 @@
 
 public class LocalFilesRoundtripTest extends TestBase {
 
-  SimpleExtension.ExtensionCollection extensions = defaultExtensionCollection;
-
-  public LocalFilesRoundtripTest() throws IOException {}
```
Comment (vbarua) [inline core/src/test/java/io/substrait/type/proto/LocalFilesRoundtripTest.java:30]: This constructor is not needed.

**Review by EpsilonPrime — APPROVED** [review]
Nice cleanup!

**Comment by vbarua** [comment]
CI is failing due to
```
/home/runner/work/substrait-java/substrait-java/core/src/test/java/io/substrait/type/proto/LocalFilesRoundtripTest.java:61: error: switch expressions are not supported in -source 8
    return switch (pathTypeCase) {
           ^
  (use -source 14 or higher to enable switch expressions)
/home/runner/work/substrait-java/substrait-java/core/src/test/java/io/substrait/type/proto/LocalFilesRoundtripTest.java:62: error: switch rules are not supported in -source 8
      case URI_PATH -> builder.pathType(FileOrFiles.PathType.URI_PATH).path("path");
```
which are in code this PR did not touch. Locally, tests run and build fine.

I've seen a number of builds fail with issues like this. In theory Jabel should be converting this code, but it doesn't seem to fire sometimes.

**Comment by vbarua** [comment]
Filed https://github.com/substrait-io/substrait-java/issues/195 for the build issue.

**Pull request merged**